lib/internal/fs/streams.js

```diff
@@ -374,9 +398,16 @@ WriteStream.prototype._writev = function(data, cb) {
     size += chunk.length;
   }
 
+  this.isPerformingIO = true;
   fs.writev(this.fd, chunks, this.pos, function(er, bytes) {
+    this.isPerformingIO = false;
+    // Tell ._destroy() that it's safe to close the fd now.
+    if (this.destroyed) return this.emit(kIoDone);
```

Comment:
This will swallow the error and deadlock a write(cb) call (i.e. never invoke the callback).

Reply:
```js
if (this.destroyed) {
  cb(er); // Don't swallow error and ensure callback is invoked.
  this.emit(kIoDone);
  return
}
```

Reply:
> This will swallow the error and deadlock a write(cb) call (i.e. never invoke the callback).

I’ve updated the code to always call the callback.